### PR TITLE
SAK-49206 Lessons deleting question should delete responses

### PR DESCRIPTION
**Pull request by jkjanetschek**
…oes delete all corresponding response entries in database

### CODE REVIEW

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java [hunk 1]
```diff
@@ -1982,4 +1982,17 @@ public void deleteLogForLessonsItem(SimplePageItem item) {
...
 	}
+
+  public void deleteQuestionResponsesForItem(SimplePageItem item) {
+		try {
+			DetachedCriteria d = DetachedCriteria.forClass(SimplePageQuestionResponse.class).add(Restrictions.eq("questionId",item.getId()));
+			DetachedCriteria d2 = DetachedCriteria.forClass(SimplePageQuestionResponseTotals.class).add(Restrictions.eq("questionId",item.getId()));
+			getHibernateTemplate().deleteAll(getHibernateTemplate().findByCriteria(d));
+			getHibernateTemplate().deleteAll(getHibernateTemplate().findByCriteria(d2));
+		} catch (DataAccessException e) {
+			log.warn("Failed to delete SimplePageQuestion responses for item {}", item.getId());
```
Comment (adrianfish) [inline lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java:1993]: I think this should be an error. It's a bad situation where we get this kind of exception. I'd also add e.toString() to the log error.

**Review by adrianfish — CHANGES_REQUESTED** [review]
I think you should emit an error to the logs, rather than a warn. We also need the e.toString() so we get the exact location in the code where it failed.

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java (hunk 1 above)
Comment (jkjanetschek) [reply lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java]: Thank you for your input! I did update the PR

**Review by adrianfish — APPROVED** [review]
Thanks!

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java (hunk 1 above)
Comment (jkjanetschek) [reply lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java]: Thank you for your input! I did update the PR

**Review by jesusmmp — APPROVED** [review]
LGTM!

**Pull request merged**